nilearn/_utils/glm.py

```diff
@@ -143,11 +143,41 @@ def _check_run_tables(run_imgs, tables_, tables_name):
     return tables_
 
 
-def z_score(pvalue):
-    """ Return the z-score corresponding to a given p-value.
+def z_score(pvalue, one_minus_pvalue=None):
+    """ Return the z-score(s) corresponding to
+    certain p-value(s) and cdfvalues(s) provided as inputs.
+
+    Parameters
+    ----------
+    pvalue: float or 1-d array shape=(n_pvalues,) computed using
+            the survival function
+
+    cdfvalue: float or 1-d array shape=(n_cdfvalues,), optional;
+              it shall take the value returned by
+              /nilearn/glm/contrasts.py::one_minus_pvalue
+              which computes the p_value using the
+              cumulative distribution function, with n_cdfvalues = n_pvalues
+
+    Returns
+    -------
+    z_scores: 1-d array shape=(n_z_scores,), with n_z_scores = n_pvalues
     """
-    pvalue = np.minimum(np.maximum(pvalue, 1.e-300), 1. - 1.e-16)
-    return norm.isf(pvalue)
+    pvalue = np.array(np.minimum(np.maximum(pvalue, 1.e-300), 1. - 1.e-16))
```

Comment:
you can use `np.clip`

Reply:
OK. I have changed both the threshold assessment of pvalue and one_minus_pvalue using np.clip.